source/index.html.md

```diff
@@ -808,6 +808,42 @@ object][ApplePay-PaymentToken] for more information.
   subsequent-in-series authorization.
 </p>
 
+Additionally, an Apple Pay authorization can be created using raw values from
+a payment token:
+
+<dl class="dl-vertical">
+  <dt>applepay[raw][pan]
+    <span class="type">[0-9]{12,19}</span>
+  </dt>
+  <dd>
+    Primary account number of card to charge.
+  </dd>
+  <dt>applepay[raw][expire_month]
+    <span class="type">[0-9]{2}</span>
+  </dt>
+  <dd>
+    Expiry month of card to charge.
+  </dd>
+  <dt>applepay[raw][expire_year]
+    <span class="type">20[0-9]{2}</span>
+  </dt>
+  <dd>
+    Expiry year of card to charge.
+  </dd>
+  <dt>applepay[raw][cryptogram]
+    <span class="type">[:base64:]{28}</span>
+  </dt>
+  <dd>
+    Online payment cryptogram.
+  </dd>
+  <dt>applepay[raw][eci]
+    <span class="type">[0-9]{1,2}</span>
```

Comment:
I stumble upon the variable length. Usually ECI is always 2, but in the Apple Pay tokens we receive it's always 1 (just hunted it down to be sure that we log it "very raw"); and [the spec](https://developer.apple.com/library/archive/documentation/PassKit/Reference/PaymentTokenJSON/PaymentTokenJSON.html) doesn't mention anything about length 🙁

* `[1-9][0-9]?`? Nah. That's likely "too regexp-y" 🙁
* `([1-9]|[1-9][0-9])`? Nah, don't like it.

Uhm, couldn't we just require it to be 2 long? PSPs doing integrations "more directly" like this, they'd likely need to make the ECI be 2 anyway ... WDYT?

Reply:
I actually thought `[0-9]{1,2}` was a good compromise :smile: However, I also like the idea about being "strict". I'll take a look the code.

Reply:
https://github.com/clearhaus/gateway-api-docs/pull/137/commits/c39552d091539cbe66ecceff8400d0f1090962cb